mat-process-api/V1/Validators/GetProcessImageRequestValidator.cs

```diff
@@ -13,10 +13,12 @@ public GetProcessImageRequestValidator()
         {
             ValidatorOptions.CascadeMode = CascadeMode.StopOnFirstFailure; // Stops the validator from going into DependentRules or deeper into any single rule in general. This is needed because it doesn't make sense to keep checking if property is empty, once it was determined it's null.
 
-            RuleFor(req => req.processRef).Must(ValidateGuid)
-               .WithMessage("You need to provide a valid process reference.");
-            RuleFor(req => req.imageId).Must(ValidateGuid)
-                .WithMessage("You need to provide a valid Image Id.");
+            RuleFor(req => req.processRef).NotNull().WithMessage("Process reference must be provided.").NotEmpty().WithMessage("Process reference must be provided.") //Added the NotNull,NotEmpty checks back to get more specific error messages
+                .Must(ValidateGuid).WithMessage("You need to provide a valid process reference.");
+            RuleFor(req => req.imageId).NotNull().WithMessage("Image Id must be provided.").NotEmpty().WithMessage("Image Id must be provided.")
+                .Must(ValidateGuid).WithMessage("You need to provide a valid Image Id.");
+            RuleFor(req => req.processType).NotNull().WithMessage("Process Type must be provided.").NotEmpty().WithMessage("Process Type must be provided.");
```

Comment:
would it be worth specifying in the NotEmpty message that the value cannot be empty? Or if same message is used then combining those two checks i.e. NotNull().NotEmpty().WithMessage("...

Reply:
Tried this `NotNull().NotEmpty().WithMessage("...`  before it didn't return the message on notnull check.
But I agree, I was thinking the exact same thing, it would be nice to have different error messages. I'll need to split the tests to do this though.

Reply:
Resolved.